neuralmonkey/decoding_function.py

```diff
@@ -5,13 +5,29 @@
 See http://arxiv.org/abs/1606.07481
 """
 from abc import ABCMeta
+from typing import Any, Dict, List, Tuple, NamedTuple
 
 import tensorflow as tf
 from neuralmonkey.nn.projection import linear
 from neuralmonkey.nn.ortho_gru_cell import OrthoGRUCell
 
+# pylint: disable=invalid-name
```

Comment:
Někde tady (třeba v dokumentaci modulu) by se hodilo popsat, jaké rozhraní má ten attention. Že má ty různé historie, které vznikají různým použitím v různých dekodérech a že má ten `LoopState`, což je místo, kde se hromadí všechny side-efekty  jeho volání: hromadí se tam historie, dá se použít třeba trackování coverage.